dao/src/main/java/org/thingsboard/server/dao/sql/JpaAbstractDao.java

```diff
@@ -60,21 +85,30 @@ public D save(TenantId tenantId, D domain) {
         }
         setSearchText(entity);
         log.debug("Saving entity {}", entity);
-        if (entity.getUuid() == null) {
+        boolean isNew = entity.getUuid() == null;
+        if (isNew) {
             UUID uuid = Uuids.timeBased();
             entity.setUuid(uuid);
             entity.setCreatedTime(Uuids.unixTimestamp(uuid));
         }
-        entity = getRepository().save(entity);
-        return DaoUtil.getData(entity);
+
+        if (preSaveAction != null) {
```

Comment:
You don't need another method.
just use an ordinary save

partitioned stuff is not related to eny antity, but for the whole dao (e.g. auditLogDao)
in JpaAbstractDao
this.isPartitioned() -- default false
this.createPartitionIfExists() -- default empty method

Override it in  JpaAuditLogDao

Reply:
I prefer current approach

Reply:
Anyway, passing an effective constant as a parameter is not good.
I am afraid we all will be forced to use more OOP and less functions :)